rx/src/test/java/net/grandcentrix/thirtyinch/rx/RxTiPresenterSubscriptionHandlerTest.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2016 grandcentrix GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.grandcentrix.thirtyinch.rx;
+
+import net.grandcentrix.thirtyinch.TiView;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import rx.observers.TestSubscriber;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.mockito.Mockito.mock;
+
+@RunWith(JUnit4.class)
+public class RxTiPresenterSubscriptionHandlerTest {
+
+    private TiMockPresenter mPresenter;
+
+    private RxTiPresenterSubscriptionHandler mSubscriptionHandler;
+
+    private TiView mView;
+
+    @Before
+    public void setUp() throws Exception {
+        mView = mock(TiView.class);
+        mPresenter = new TiMockPresenter();
+        mSubscriptionHandler = new RxTiPresenterSubscriptionHandler(mPresenter);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        mPresenter = null;
+        mView = null;
+        mSubscriptionHandler = null;
+    }
+
+    @Test(expected = AssertionError.class)
+    public void tesManageViewSubscription_DetachBeforeAttach_ShouldThrowAssertError()
+            throws Exception {
+        mPresenter.create();
+        TestSubscriber<Integer> testSubscriber = new TestSubscriber<>();
+
+        mSubscriptionHandler.manageViewSubscription(testSubscriber);
+        mPresenter.detachView();
+
+        testSubscriber.assertUnsubscribed();
+    }
+
+    @Test(expected = IllegalStateException.class)
```

Comment:
I prefer the previous used try catch solution because the exception message was checked too